cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/store/SourceControlMetadataStoreTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2024 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.store;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import io.cdap.cdap.common.ApplicationNotFoundException;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.proto.sourcecontrol.SourceControlMeta;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import java.time.Instant;
+import org.junit.Before;
+import org.junit.Test;
+
+public abstract class SourceControlMetadataStoreTest {
+
+  private static final String NAMESPACE = "testNamespace";
+  private static final String COMMIT_ID = "testCommitId";
+  private static final String SPEC_HASH = "testSpecHash";
+  private static final Instant LAST_MODIFIED = Instant.now();
+  private static final String NAME = "testName";
+  private static final int INITIAL_NUMBER_OF_ROWS = 1;
+  private static final int UPDATED_NUMBER_OF_ROWS = 2;
+  private static final ApplicationId APP_ID = new ApplicationId(NAMESPACE, NAME);
+  private static final SourceControlMeta SOURCE_CONTROL_META = new SourceControlMeta(SPEC_HASH,
+      COMMIT_ID, LAST_MODIFIED);
+
+  protected static TransactionRunner transactionRunner;
+
+  @Before
+  public void before() {
+    TransactionRunners.run(transactionRunner, context -> {
+      SourceControlMetadataStore store = SourceControlMetadataStore.create(context);
+      store.deleteNamespaceSourceControlMetadataTable();
+    });
+
+    TransactionRunners.run(transactionRunner, context -> {
+      SourceControlMetadataStore store = SourceControlMetadataStore.create(context);
+      store.setNamespaceSourceControlMeta(APP_ID, SOURCE_CONTROL_META);
+    });
+  }
+
+  @Test
+  public void testSetNamespaceSourceControlMeta() throws Exception {
+    TransactionRunners.run(transactionRunner, context -> {
+      SourceControlMetadataStore store = SourceControlMetadataStore.create(context);
+      store.setNamespaceSourceControlMeta(new ApplicationId(NAMESPACE, "test2"), null);
+      long count = store.countNamespaceSourceControlMetadataRecords();
```

Comment:
why you need a count method ? you should just have a list method and then verify the list with the input records

Reply:
I have removed the count test and instead tested only the content